File: tests/fixtures/codegen/inputs/standalone-expected/typescript/runtime/queries/items/TupleUpdate.ts
import type { TupleUpdateParams } from "../../../types/queries/items/TupleUpdate.js";
export type { TupleUpdateParams } from "../../../types/queries/items/TupleUpdate.js";
import * as pgsid from "./pgsid/queryable.js";
export { QueryValidationError } from "./pgsid/queryable.js";
import * as _jsonSchemas from "../../jsonschemas/index.js";
export const tupleUpdateSql = "UPDATE items SET (payload, unchecked) = ($1, $2);";
export function isTupleUpdateParamsFirstPublicItemsPayload(value: unknown): value is TupleUpdateParams["first"] {
    return _jsonSchemas.isEvent(value);
}
export function validateTupleUpdateParamsFirstPublicItemsPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEvent(value);
}
export async function tupleUpdate(db: pgsid.Queryable, params: TupleUpdateParams): Promise<void> {
    const jsonInput1 = JSON.stringify(params["first"]);
    if (jsonInput1 === undefined)
        throw new pgsid.QueryValidationError("TupleUpdate", "first", [{ path: [], keyword: "json", expected: "JSON value", received: "undefined", message: "Value has no JSON representation" }]);
    const jsonInput1Value = jsonInput1 === null ? null : JSON.parse(jsonInput1);
    const jsonInput2 = params["second"] === null ? null : JSON.stringify(params["second"]);
    if (jsonInput1 !== null) {
        if (!isTupleUpdateParamsFirstPublicItemsPayload(jsonInput1Value))
            throw new pgsid.QueryValidationError("TupleUpdate", "first", validateTupleUpdateParamsFirstPublicItemsPayload(jsonInput1Value).issues);
    }
    await db.query(tupleUpdateSql, [jsonInput1, jsonInput2]);
}

File: tests/fixtures/codegen/inputs/pointers-expected/queries/items/tupleupdate.go
package items

import (
	"context"
	jsonschemas "example.com/pgsid-validation/generated/jsonschemas"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
)

const TupleUpdateSQL = "UPDATE items SET (payload, unchecked) = ($1, $2);"

type TupleUpdateParams struct {
	First  jsonschemas.Event  `db:"first"`
	Second *jsonschemas.Event `db:"second"`
}

func (q *Queries) TupleUpdate(ctx context.Context, params TupleUpdateParams) error {
	jsonInput1, err := pgsidpgx.ValidateJSONInput(params.First, "pgsid:///jsonschemas/Event.json#", "TupleUpdate", "first", false)
	if err != nil {
		return err
	}
	_, err = q.db.Exec(ctx, TupleUpdateSQL, jsonInput1, params.Second)
	return err
}
